ipa-core/src/net/server/handlers/query/mod.rs

```diff
@@ -55,10 +58,18 @@ pub fn h2h_router(transport: MpcHttpTransport) -> Router {
 /// Construct router for shard-to-shard communications similar to [`h2h_router`].
 pub fn s2s_router(transport: ShardHttpTransport) -> Router {
     Router::new()
-        .merge(prepare::router(transport.inner_transport))
+        .merge(prepare::router(Arc::clone(&transport.inner_transport)))
+        .merge(results::router(transport.inner_transport))
         .layer(layer_fn(HelperAuthentication::<_, Shard>::new))
 }
 
+/// Client-to-shard routes. There are only a few cases where we expect parties
+/// to talk to individual shards. Input submission is one of them. This path does
+/// not require cert authentication.
+pub fn c2s_router(transport: &ShardHttpTransport) -> Router {
+    Router::new().merge(input::router(Arc::clone(&transport.inner_transport)))
```

Comment:
Why not leave input in the MPC routes, minimizing the number of changes?

I don't see advantage on sending this to the shard port instead.

Reply:
good catch, this is actually not required - we don't want clients to get shard clients. I removed it and that simplified this change by quite a bit